src/frontends/pytorch/src/op/index_add.cpp

```diff
@@ -0,0 +1,105 @@
+    auto slice_step = context.mark_node(std::make_shared<v3::Broadcast>(const_one, src_rank));
+    auto dim_1d = context.mark_node(std::make_shared<v3::Broadcast>(dim, const_one, BroadcastType::BIDIRECTIONAL));
+    slice_end =
+        context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(slice_end,
+                                                                       dim_1d,
+                                                                       const_one,
+                                                                       const_zero,
+                                                                       v12::ScatterElementsUpdate::Reduction::NONE));
+    auto new_shape = context.mark_node(std::make_shared<v8::Slice>(input, slice_start, slice_end, slice_step, axes));
+    new_shape = context.mark_node(std::make_shared<v3::ShapeOf>(new_shape, element::i32));
+    auto src_ =
+        context.mark_node(std::make_shared<v3::Broadcast>(src_broadcasted, new_shape, BroadcastType::BIDIRECTIONAL));
+    auto src_input_dtype = context.mark_node(std::make_shared<v1::ConvertLike>(src_, input));
+    // brodcast index to input rank size
+    src_rank = context.mark_node(std::make_shared<v3::ShapeOf>(new_shape, element::i32));
+    auto new_index_shape = context.mark_node(std::make_shared<v3::Broadcast>(const_one, src_rank));
+    auto const_minus_one = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {-1}));
+    new_index_shape =
+        context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(new_index_shape,
+                                                                       dim_1d,
+                                                                       const_minus_one,
+                                                                       const_zero,
+                                                                       v12::ScatterElementsUpdate::Reduction::NONE));
+    // precerve indicies location for spicifc dim
+    auto reshaped_index = context.mark_node(std::make_shared<v1::Reshape>(index, new_index_shape, false));
+    auto broadcasted_index =
+        context.mark_node(std::make_shared<v3::Broadcast>(reshaped_index, new_shape, BroadcastType::BIDIRECTIONAL));
+    auto scatter_result =
+        context.mark_node(std::make_shared<v12::ScatterElementsUpdate>(input,
+                                                                       broadcasted_index,
+                                                                       src_,
+                                                                       dim,
+                                                                       v12::ScatterElementsUpdate::Reduction::SUM));
+    if (!context.input_is_none(5)) {
+        context.mutate_input(5, scatter_result);
+    }
+    return {scatter_result};
+}
+
+OutputVector translate_index_add_(const NodeContext& context) {
```

Comment:
This can be done with `inplace_op` decorator, right?

Reply:
reused